### DISCUSSION TITLE
Can gmtsar calculate the backscatter?

### DISCUSSION THREAD

**Discussion by kjz1997** [gmtsar/gmtsar#997] [General]
hello Dr
I want to know if gmtsar can be used to calculate backscatter

**Comment by Xiaohua-Eric-Xu** [comment]
There are ways to create the needed amplitude image, e.g., with slc2amp.csh and then geocode your product by running dem2topo_ra.csh and proj_ra2ll.csh. You may need to change the filter inside. This amplitude has a factor of 2.5e-7 multiplied twice when reading in float/int data. You may need to convert it back and use the amp info in the xml file to get the real backscatter.

**Comment by kpzt-sven** [comment]
Indeed it should be doable, but I have not figured out the exact routine yet. Executed commands:

- make_s1a_tops S1A_IW_SLC__1SDV_20200524T043423_20200524T043451_032702_03C9AF_3580.SAFE/annotation/s1a-iw2-slc-vv-20200524t043424-20200524t043449-032702-03c9af-005.xml S1A_IW_SLC__1SDV_20200524T043423_20200524T043451_032702_03C9AF_3580.SAFE/measurement/s1a-iw2-slc-vv-20200524t043424-20200524t043449-032702-03c9af-005.tiff s1a-iw2-slc-vv_3580 0
- or make_slc_s1a S1A_IW_SLC__1SDV_20200524T043423_20200524T043451_032702_03C9AF_3580.SAFE/annotation/s1a-iw2-slc-vv-20200524t043424-20200524t043449-032702-03c9af-005.xml S1A_IW_SLC__1SDV_20200524T043423_20200524T043451_032702_03C9AF_3580.SAFE/measurement/s1a-iw2-slc-vv-20200524t043424-20200524t043449-032702-03c9af-005.tiff s1a-iw2-slc-vv_3580_2
- make_dem.csh 25.294 27.519 57.398 59.112 2
- dem2topo_ra.csh s1a-iw2-slc-vv_3580_2.PRM dem.grd (Creates trans.dat - https://gmtsar.github.io/documentation/Projecting_To_LatitudeLongitude.html#projecting-to-latitudelongitude)
- slc2amp.csh s1a-iw2-slc-vv_3580_2.PRM 2 amp.grd
- proj_ra2ll.csh trans.dat amp.grd amp_ll.grd 100
- gmt grdconvert amp_ll.grd amp.tif=gd:GTiff (Does not do correct geocoordinates exactly, but closer)

However my question is the outcome of the slc2amp.csh. The values are miniscule, close to zero, there would be a lot of precision gone when dividing by 2.5e-7 twice afterwards. Is there any other way to get the geocoordinated data either separate I and Q or some other format?
Finally, what is the correct way to convert the resulting grd to a GeoTIFF for example ?

**Reply by Xiaohua-Eric-Xu** [reply]
Hi @kjz1997, the way to easily convert grd to geotiff is to use `gdal_translate`.  To get I and Q directly, you may reads in the SLC data directly as it is simply a binary chunk of data with short integers (int16) in the format of real, imag, real imag. The num_lines and num_rng_bins denote lines and columns of the chunk.  Either python or matlab or c could easily get this read and write into any format you like :).  Do note slc2amp.csh actually convolves a gaussian filter, and you may easily multiply back the factor with `gmt grdmath`.

**Reply by kpzt-sven** [reply]
Thank you for the response. Indeed my further ideas are aiming the same direction. However there are some caveats:

- Example data given above has 3 more lines in the SLC file than in the PRM, the file size is (3 + num_lines) x num_rng_bins x 2 (e.g. sizeof int16)
- After calculating intensity on SLC real imag pixels how does further tools associate it with a PRM file if saving it into a grid format?

**Reply by Xiaohua-Eric-Xu** [reply]
The num_lines is the number of lines that will be calculated. It is set to be divisible by 4 considering the actual resolution of DEM is usually larger than the sampling rate. This is sort of needed in case one needs to interpolate the DEM to calculate topographic phase. The grid information could be get with `gmt grdinfo`.

**Reply by kpzt-sven** [reply]
Thank you for the answers. However I disagree with the fact that it is OK that there is such an ambiguity in the system where SLC data is somehow influenced by DEM file which is fetched completely separate from the radar dataset... 

For example dem2topo_ra.csh logs:
`Working over 0/26164/0/12196 ...`
Which is exactly the sizes stated in the PRM...

Nevertheless I managed to find the solution thanks to your help:
- `make_s1a_tops <XML> <TIFF> 1`
- `make_dem.csh <bounding coords> <1/2>`
- `dem2topo_ra.csh <PRM> <dem.grd>`
- SLC 2 intensity script (based on @Xiaohua-Eric-Xu guidance above) - Read only the amount of lines described in the PRM
- `proj_ra2ll.csh <trans.dat> <intensity.grd - SLC 2 intensity script output> <result_ll.grd> 100`
- `gdalwarp -s_srs '+proj=longlat +datum=WGS84 +lon_wrap=180' -t_srs '+proj=longlat +datum=WGS84' -dstnodata 0 <result_ll.grd> <result_ll.tif>`

Based on one's requirements one could/would add the LED update via ext_orb_s1a and merging/slicing too....

**Reply by Xiaohua-Eric-Xu** [reply]
I understand that one would like to have all cases precisely defined, yet the software was designed to utilize precise orbit and speeded decimation to generate interferograms in a faster fashion. The numbers could be controlled by changing some of the numbers in the scripts or run step by step processing with customized parameters. Do note that for proj_ra2ll.csh the spatial sampling in the end will be a quarter of that number.
